modes/beginend/evil-collection-beginend.el

```diff
@@ -55,100 +79,55 @@
                                           beginend-notmuch-search-mode-map
                                           beginend-recentf-dialog-mode-map))
 
+
+(defmacro evil-beginend-goto-beginning-motion (ec-mode-name)
+  "Macro to define new Evil motion that will use the corresponding
+beginend-goto-beginning function for MODE-NAME when count is not
+provided for the motion; otherwise behave like `evil-goto-first-line'
+if count is provided.
+
+This will also associate the regular Evil `gg' keybinding with the
+newly defined motion."
+  (let ((motion-name (intern (format "evil-beginend-%s-goto-beginning" ec-mode-name)))
+        (beginend-beginning-fn-name (intern (format "beginend-%s-goto-beginning" ec-mode-name)))
+        (beginend-map-name (intern (format "beginend-%s-map" ec-mode-name))))
+    `(progn
+       (evil-define-motion ,motion-name (count)
+         :jump t
+         :type line
+         (if count
+             (evil-goto-first-line count)
+           (,beginend-beginning-fn-name)))
+       (evil-collection-define-key 'normal ',beginend-map-name
+         "gg" ',motion-name))))
+
+(defmacro evil-beginend-goto-end-motion (ec-mode-name)
```

Comment:
nitpicking: `evil-beginend--define-goto-end-motion` looks more meaningful. Two `-`s since is a private macro.